lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -846,17 +846,20 @@ def associations_for_user_or_attribute(user_or_attribute, options)
     def build_accessible_object_scope(associations, options = {})
       permitting_oas = PolicyElement.where(id: associations.pluck(:object_attribute_id))
 
+      scopes = []
+
       # Direct scope: the set of objects on which the operator is directly assigned
-      direct_scope = permitting_oas.where(type: class_for_type('object').name)
-      # Indirect scope: the set of objects which the operator can access via ancestral hierarchy
-      indirect_scope = Assignment.ancestors_of(permitting_oas).where(type: class_for_type('object').name)
+      scopes << permitting_oas.where(type: class_for_type('object').name)
 
-      if inclusion = options[:includes]
-        direct_scope = build_inclusion_scope(direct_scope, options[:key], inclusion)
-        indirect_scope = build_inclusion_scope(indirect_scope, options[:key], inclusion)
+      # Indirect scope: the set of objects which the operator can access via ancestral hierarchy
+      unless options[:direct_only]
+        scopes << Assignment.ancestors_of(permitting_oas).where(type: class_for_type('object').name)
       end
 
-      direct_scope | indirect_scope
+      inclusion = options[:includes]
+      scopes.map! { |s| build_inclusion_scope(s, options[:key], inclusion) } if inclusion
+
+      scopes.reduce(&:|).to_a
```

Comment:
Think for a user of this interface external to the PM, a more generally useful option might be something like `depth`, where with some external knowledge of the shape of the object hierarchy, a system could limit recursion to some specified depth, `direct` being depth 0 or 1 depending on how the underlying recursive query gem likes to count.  Don't think that is necessary for the immediate use case, though